cli/main.rs

```diff
@@ -558,9 +558,15 @@ async fn eval_command(
   program_state.file_fetcher.insert_cached(file);
   debug!("main_module {}", &main_module);
   worker.execute_module(&main_module).await?;
-  worker.execute("window.dispatchEvent(new Event('load'))")?;
+  worker.execute_script(
+    "[native code]",
+    "window.dispatchEvent(new Event('load'))",
```

Comment:
What do you think of something like [`std::file!()`](https://doc.rust-lang.org/std/macro.file.html) for JS inlined in rust?
```suggestion
  runtime.execute_script(&format!("[deno:{}].js", std::file!()), &init_src)?;
```
![image](https://user-images.githubusercontent.com/29990554/122626520-f219f980-d0a2-11eb-917c-2bd1e043f883.png)

Reply:
Yeah, that was my first thought but I didn't want to over do it. 

I'll be fine even with copy-pasting the strings, don't think a couple dozen call sites across 3 modules need a macro

Reply:
I think it's a good idea maybe something like 
```
format!("[deno-internal:{}]", std::file!())
```

Reply:
execute_script could even become a macro, so this code wasn't repeated everywhere...

Reply:
> ```
> format!("[deno-internal:{}]", std::file!())
> ```

It can just be `deno:`, the situation matches exactly how we use this protocol: `deno:<repo-relative path to source file>`.

Reply:
Well to make it consistent with code in extensions it should be `deno:cli/main.rs` and I think that's a good solution - it will be obvious that this code is triggered from Rust. 

So I guess full suggestion would be to:
```rust
worker.execute_script(
  format!("deno:", std::file!()),
  "window.dispatchEvent(new Event('load'))"
);
```

Let me try to implement that and see how it goes.

> execute_script could even become a macro, so this code wasn't repeated everywhere...

-1 on this one - `Worker` and `WebWorker` are in `runtime` crate that is used by 3rd parties that embed deno - it's not necessary true they want to have such filename, instead they might want to use custom name.

Reply:
> Well to make it consistent with code in extensions it should be `deno:cli/main.rs`

There should be some kind of separator at the end so the path is cut off from wrongly contextualised line/column numbers in the event of an error. That's why I think square brackets are a good idea:
![image](https://user-images.githubusercontent.com/29990554/122650962-b0826080-d12d-11eb-992e-24ce93586d9f.png)
![image](https://user-images.githubusercontent.com/29990554/122650974-c55ef400-d12d-11eb-94d0-fadac2af6c83.png)

Reply:
I just checked and `std::file!()` appends leading slash:
```
let a = format!("deno:/{}", std::file!());
eprintln!("file {}", a);
```
```
file deno:/cli/main.rs
```

So I think I'll just go with strings for now.

> There should be some kind of separator at the end so the path is cut off from wrongly contextualised line/column numbers in the event of an error. That's why I think square brackets are a good idea:

I'm open to that, though it doesn't bother me that much - in either situation you won't be able to lookup this code using column and line anyway (line will always be 1, as we only execute onliners).

Reply:
> I just checked and `std::file!()` appends leading slash:
> 
> ```
> let a = format!("deno:/{}", std::file!());
> ```

Looks like the leading slash is yours? 😅

Reply:
🤦

Reply:
Alright, I reworked it to use `std::file!()` haven't added curly braces. I'm open to doing that, @ry wanna break the tie on this one?

Reply:
> I'm open to that, though it doesn't bother me that much - in either situation you won't be able to lookup this code using column and line anyway (line will always be 1, as we only execute onliners).

That's why I think it shouldn't mislead with a connected line number.

However, I think we can give useful location info. I didn't suggest it because it seemed like a long line to repeat but we can do it with a macro:
```rust
macro_rules! located_script_name {
  () => {
      format!("[deno:{}:{}:{}]", std::file!(), std::line!(), std::column!());
  };
}

// ...

worker.execute_script(
  &located_script_name!(),
  "window.dispatchEvent(new Event('load'))"
);
```
The location points to the initializer of the script name but it's genuinely useful.

Reply:
Thanks, I added this macro to `deno_core` and reexported it from there so both `deno_runtime` and `deno_cli` crates can use it.